docs/structured-streaming-state-data-source.md

```diff
@@ -0,0 +1,248 @@
+---
```

Comment:
Could we create a streaming specific sub-directory under `docs` ? might be better to put all streaming doc pages under one place ?

Reply:
It may need broader change as there are links for images/js/css. I guess sql is really something we want to extract out if we can have sub-directories.